apps/crn-server/test/data-providers/contentful/research-output.data-provider.test.ts

```diff
@@ -0,0 +1,1541 @@
+            filter: {
+              teamId: 'team-0',
+            },
+            includeDrafts: true,
+          });
+
+          expect(contentfulGraphqlClientMock.request).toHaveBeenCalledWith(
+            expect.anything(),
+            expect.objectContaining({
+              where: {
+                teams: {
+                  sys: {
+                    id: 'team-0',
+                  },
+                },
+              },
+            }),
+          );
+        });
+
+        test('Should pass the parameters to filter by working group as expected', async () => {
+          await researchOutputDataProvider.fetch({
+            take: 13,
+            skip: 7,
+            filter: {
+              workingGroupId: 'wg-0',
+            },
+          });
+
+          expect(contentfulGraphqlClientMock.request).toHaveBeenCalledWith(
+            expect.anything(),
+            expect.objectContaining({
+              where: {
+                workingGroup: {
+                  sys: {
+                    id: 'wg-0',
+                  },
+                },
+              },
+            }),
+          );
+        });
+
+        test('Should pass the parameters to filter by document type as expected', async () => {
+          await researchOutputDataProvider.fetch({
+            take: 13,
+            skip: 7,
+            filter: {
+              documentType: 'Preprint',
+            },
+          });
+
+          expect(contentfulGraphqlClientMock.request).toHaveBeenCalledWith(
+            expect.anything(),
+            expect.objectContaining({
+              where: {
+                documentType: 'Preprint',
+              },
+            }),
+          );
+        });
+
+        test('Should pass the parameters to filter by document type as expected when document type is an array', async () => {
+          await researchOutputDataProvider.fetch({
+            take: 13,
+            skip: 7,
+            filter: {
+              documentType: ['Preprint', 'Report'],
+            },
+          });
+
+          expect(contentfulGraphqlClientMock.request).toHaveBeenCalledWith(
+            expect.anything(),
+            expect.objectContaining({
+              where: {
+                documentType_in: ['Preprint', 'Report'],
+              },
+            }),
+          );
+        });
+      });
+    });
+  });
+
+  describe('fetchById', () => {
+    test('can call from mock server', async () => {
+      await researchOutputDataProviderMockGraphql.fetchById('1');
+    });
+
+    test('throws a not found error if query does not return a result', async () => {
+      contentfulGraphqlClientMock.request.mockResolvedValue({
+        researchOutputs: null,
+      });
+      await expect(researchOutputDataProvider.fetchById('1')).rejects.toThrow(
+        'Not Found',
+      );
+    });
+
+    test('looks for published version first, and then draft if there is no published version', async () => {
```

Comment:
As I understand this is done because our API does not actually behave differently whether you ask for a draft or published RO? In which case I wonder why you went with this approach instead of simply asking for the resource with  the preview flag set to true right from the start? If you were switching between the different tokens (as suggested further up) then that would make sense but at this stage i would say it's unnecessary

Reply:
If an ASAP admin makes a change to a published output in the CMS, but then _doesn't_ publish the changes then this makes sure that the published version is still returned if that output is requested via the hub until such a time as they publish.